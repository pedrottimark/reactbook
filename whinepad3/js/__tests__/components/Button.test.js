import React from 'react';
import TestUtils from 'react-addons-test-utils';
import renderer from 'react-test-renderer';

import Button from '../../source/components/Button';

// For TestUtils because Button is a functional component.
class Wrap extends React.Component {
  render() {
    return <div>{this.props.children}</div>;
  }
}

describe('Button with href prop', () => {

  it('renders <a>', () => {
    const tree = renderer.create(
      <Button href="#">Hello</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Button with no href prop', () => {

  it('renders <button>', () => {
    const tree = renderer.create(
      <Button>Hello</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Button', () => {

  it('appends className prop to class attribute', () => {
    const tree = renderer.create(
      <Button className="good bye">Hello</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders onClick prop as attribute', () => {
    const tree = renderer.create(
      <Button onClick={() => {}}>Hello</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick when people click', () => {
    const onClick = jest.fn();
    const component = TestUtils.renderIntoDocument(
      <Wrap><Button onClick={onClick}>Hello</Button></Wrap>
    );
    const button = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
    TestUtils.Simulate.click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

});
