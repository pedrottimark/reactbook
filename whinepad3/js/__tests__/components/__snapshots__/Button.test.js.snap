exports[`Button appends className prop to class attribute 1`] = `
<button
  className="Button good bye">
  Hello
</button>
`;

exports[`Button with href prop renders <a> 1`] = `
<a
  className="Button"
  href="#">
  Hello
</a>
`;

exports[`Button with no href prop renders <button> 1`] = `
<button
  className="Button">
  Hello
</button>
`;

exports[`Button renders onClick prop as attribute 1`] = `
<button
  className="Button"
  onClick={[Function]}>
  Hello
</button>
`;
